src/distilabel/llm/openai.py

```diff
@@ -184,3 +194,144 @@ def _generate(
+                If `None`, the `OPENAI_API_KEY` environment variable will be used. Defaults to `None`.
+            max_new_tokens (int, optional): the maximum number of tokens to be generated.
+                Defaults to 128.
+            frequency_penalty (float, optional): the frequency penalty to be used for generation.
+                Defaults to 0.0.
+            presence_penalty (float, optional): the presence penalty to be used for generation.
+                Defaults to 0.0.
+            temperature (float, optional): the temperature to be used for generation.
+                Defaults to 1.0.
+            top_p (float, optional): the top-p value to be used for generation.
+                Defaults to 1.0.
+            num_threads (Union[int, None], optional): the number of threads to be used
+                for parallel generation. If `None`, no parallel generation will be performed.
+                Defaults to `None`.
+            prompt_format (Union[SupportedFormats, None], optional): the format to be used
+                for the prompt. If `None`, the default format of the task will be used, available
+                formats are `openai`, `chatml`, `llama2`, `zephyr`, and `default`. Defaults to `None`,
+                but `default` (concatenation of `system_prompt` and `formatted_prompt` with a line-break)
+                will be used if no `prompt_formatting_fn` is provided.
+            prompt_formatting_fn (Union[Callable[..., str], None], optional): a function to be
+                applied to the prompt before generation. If `None`, no formatting will be applied.
+                Defaults to `None`.
+
+        Raises:
+            AssertionError: if the provided `model` is not available in your OpenAI account.
+            AssertionError: if the provided `model` does not support JSON input.
+
+        Examples:
+            >>> from distilabel.tasks import TextGenerationTask
+            >>> from distilabel.llm import JSONOpenAILLM
+            >>> llm = OpenAILLM(task=TextGenerationTask())
+            >>> llm.generate([{"input": "json for 'What's the capital of Spain?'"}])
+        """
+        super().__init__(
+            task,
+            model=model,
+            client=client,
+            api_key=api_key,
+            max_new_tokens=max_new_tokens,
+            frequency_penalty=frequency_penalty,
+            presence_penalty=presence_penalty,
+            temperature=temperature,
+            top_p=top_p,
+            num_threads=num_threads,
+            prompt_format=prompt_format,
+            prompt_formatting_fn=prompt_formatting_fn,
+        )
+        self.json_supporting_models = [
+            "gpt-4-0125-preview",
+            "gpt-4-turbo-preview",
+            "gpt-4-1106-preview",
+            "gpt-3.5-turbo-1106",
+        ]
+        assert model in self.json_supporting_models, f"Provided `model` does not support JSON input, \
+            available models are {self.json_supporting_models}"
```

Comment:
Meaning the same property as `available_models` within `OpenAILLM` but with a slight filtering on the proprietary models offered by OpenAI

Reply:
I initially went for the separate approach because the account is asserted, then JSON compatibility is asserted. 
I can of course just log in the `available_models` method if there are no JSON compatible model in the account's models.

Reply:
Hmm fair I get your point now :/ Then you can leave it as is for the moment, and if any issue arises we can revisit, but the current is fine too!